### ISSUE TITLE
[Bug] spawn_blocking: use-after-free in detached thread fallback path

### ISSUE THREAD

**Issue by Coldwings** [Coldwings/Elio#235]
## Bug Description

In `include/elio/runtime/spawn_blocking.hpp`, the `blocking_awaitable::await_suspend()` function captures a raw pointer `state` to `state_` (which lives in the coroutine frame) and passes it to a work lambda that executes on a blocking pool thread or detached thread. If the awaiting coroutine is destroyed before the blocking task completes (e.g., during scheduler shutdown), the coroutine frame and `state_` are freed, but the blocking thread continues executing and writes to the freed memory via `state->completed`, `state->result`, or `state->exception`. This is a write-after-free.

Additionally, the fallback path at line 65 calls `caller.resume()` directly on the blocking thread with only a `!caller.done()` check, which does not protect against a destroyed handle (calling any method on a destroyed handle is UB).

## Environment

- **Elio Version**: main branch (commit 5e75710)
- **Compiler**: GCC 12+ / Clang 15+
- **OS**: Linux
- **Build Type**: All (Debug, Release, ASAN, TSAN)

## Steps to Reproduce

1. Configure a `blocking_pool` with `num_threads_ == 0` (detached thread mode) OR trigger `submit()` rejection during shutdown
2. Call `spawn_blocking()` from a coroutine
3. The work lambda is spawned as a detached thread (line 81)
4. Before the detached thread completes, destroy the awaiting coroutine (e.g., via `when_any` loser cleanup, cancellation, or `handle.destroy()`)
5. The detached thread writes to freed memory

### Minimal Reproducible Example

```cpp
#include <elio/elio.hpp>
#include <chrono>
#include <thread>

using namespace elio;

coro::task<int> slow_blocking_task() {
    // This will spawn a detached thread if blocking_pool is full or shutting down
    int result = co_await spawn_blocking([]() {
        std::this_thread::sleep_for(std::chrono::seconds(2));
        return 42;
    });
    co_return result;
}

coro::task<void> trigger_uaf() {
    auto task = slow_blocking_task();
    
    // Start the task
    auto handle = task.start();
    
    // Give time for spawn_blocking to submit work
    co_await time::sleep_for(std::chrono::milliseconds(100));
    
    // Destroy the coroutine frame while blocking task is still running
    handle.destroy();  // state_ is now freed
    
    // Detached thread continues and writes to state_->result
    // This is a use-after-free!
    
    co_return;
}
```

## Expected Behavior

The `blocking_awaitable` should ensure that `state_` outlives the work lambda, even if the awaiting coroutine is destroyed. Similarly, `caller.resume()` should only be called if the caller is still alive.

### Write-After-Free

When the awaiting coroutine is destroyed:
1. The coroutine frame (containing `blocking_awaitable` and `state_`) is deallocated
2. The work lambda (captured `state` as raw pointer) continues executing on the blocking/detached thread
3. The thread writes to `state->completed` (line 53), `state->result` (line 55), or `state->exception` (line 58)
4. **This writes to freed memory**

### UB on Destroyed Handle

Line 64-66:
```cpp
} else if (caller && !caller.done()) {
    caller.resume();
}
```

The `!caller.done()` check is meaningless for a destroyed handle. Calling any method on a destroyed `coroutine_handle` is undefined behavior per the C++ standard.

## Root Cause

1. **Raw pointer capture**: Line 40 captures `auto* state = &state_;` and passes it to the lambda (line 49). The state lives in the coroutine frame with no lifetime protection.

2. **Detached thread fallback**: Line 81 spawns `std::thread(std::move(work)).detach()`, which is completely untracked. The `blocking_pool::shutdown()` only joins pooled threads, not detached ones.

3. **Shutdown ordering**: While `scheduler::shutdown_force()` correctly shuts down the blocking pool before destroying coroutine frames, detached threads spawned just before shutdown are not tracked and can outlive the frames.

### Primary Path (blocking pool)
Normally safe because `blocking_pool::shutdown()` joins all pooled threads before `drain_remaining_tasks()` destroys frames. However, if `submit()` is rejected during shutdown (line 76 returns false), the code falls back to the detached thread path.

### Fallback Path (detached thread)
**Unsafe**. Detached threads are not joined by `shutdown()` and can write to `state_` after the coroutine frame is destroyed.

## Additional Context

- **Trigger Condition**: Requires `blocking_pool::submit()` to be rejected (pool shutting down) OR `num_threads_ == 0` (detached mode)
- **Shutdown Safety**: The main blocking pool path is safe due to careful shutdown ordering in `scheduler::shutdown_force()` (blocking pool first, then workers, then frame destruction)
- **Invalid Guard**: The `!caller.done()` check at line 64 provides a false sense of safety but does not prevent UB on destroyed handles
- **Likelihood**: Low in normal operation, but possible during shutdown or with custom scheduler lifetime management

## Related Issues

None

## Possible Solution

Replace the raw `state` pointer with a `std::shared_ptr<blocking_state<T>>` allocated on the heap. The lambda should capture it by value so the state outlives the coroutine frame.

For the `caller` safety, add an atomic flag in the shared state that the destructor sets to indicate the caller is no longer alive. Check this flag before calling `resume()`.

Example fix:
```cpp
template<typename T>
struct blocking_state {
    std::optional<T> result;
    std::exception_ptr exception;
    std::atomic<bool> caller_alive{true};  // New field
};

void await_suspend(std::coroutine_handle<> caller) {
    auto state = std::make_shared<blocking_state<T>>();
    state_ = state;  // Store in awaitable for destructor to access
    
    std::function<void()> work = [state, caller, sched, f = std::move(func_)]() mutable {
        try {
            if constexpr (std::is_void_v<T>) {
                f();
                state->completed = true;
            } else {
                state->result.emplace(f());
            }
        } catch (...) {
            state->exception = std::current_exception();
        }
        
        if (state->caller_alive.load(std::memory_order_acquire)) {
            if (sched && sched->is_running()) {
                sched->spawn(caller);
            } else if (caller && !caller.done()) {
                caller.resume();
            }
        }
    };
    // ... submit logic
}

~blocking_awaitable() {
    if (state_) {
        state_->caller_alive.store(false, std::memory_order_release);
    }
}
```

## Checklist

- [x] I have searched existing issues to avoid duplicates
- [x] I have provided all requested information
- [x] I have included a minimal reproducible example
- [x] I have tested with the latest version of Elio
- [x] I have included sanitizer output (N/A - design issue identified via code review)
